### PR TITLE
Only grant SID Winsta Access when refreshing access token

### PR DESCRIPTION
**Pull request by petemoore**
Bugzilla Bug: [1815711](https://bugzilla.mozilla.org/show_bug.cgi?id=1815711)

This is a partial fix for #6058. It should workaround the issue for the specific case raised, but will not work if custom `osGroups` entries are specified in the task payload.

### CODE REVIEW

**Comment by petemoore** [comment]
Note, bob has created a firefox-ci task which lists out part of the process access token here:

https://firefox-ci-tc.services.mozilla.com/tasks/PnVQXgGdSQiwfMkgRCBEng/runs/0/logs/public/logs/live.log

I'm seeing if I can replicate in community cluster. Current work in progress here: https://community-tc.services.mozilla.com/tasks/T1sMCOYpSFWgBklMd69BkQ

It will probably take a few iterations to get right, since e.g. mozilla-build is probably not installed on the community Windows Server 2022 machine images that the workers use...

**Comment by petemoore** [comment]
Since that task definition won't live forever, here is its definition:

```
provisionerId: gecko-t
workerType: win11-64-2009
taskQueueId: gecko-t/win11-64-2009
schedulerId: gecko-level-1
projectId: none
taskGroupId: JrG9cz9SRoKzxkIJPcB0ZA
dependencies:
  - CPxC2iUPScSXzziXbz035Q
  - HkQNjaN7SmmowDJeol8FCg
  - ewVFJ9njQE2zW_vqd50QxQ
requires: all-completed
routes:
  - tc-treeherder.v2.try.5976867c5914929594c7c0d076167ebf254c1695
priority: very-low
retries: 5
created: '2023-04-26T07:11:09.708Z'
deadline: '2023-04-27T07:11:09.708Z'
expires: '2023-05-26T07:11:09.708Z'
scopes: []
payload:
  env:
    PYTHON: c:\mozilla-build\python3\python3.exe
    MOZ_FETCHES: >-
      [{"artifact": "public/build/fix-stacks.tar.zst", "extract": true, "task":
      "HkQNjaN7SmmowDJeol8FCg"}, {"artifact":
      "public/build/minidump-stackwalk.tar.zst", "extract": true, "task":
      "CPxC2iUPScSXzziXbz035Q"}]
    TRY_SELECTOR: fuzzy
    MOZ_SCM_LEVEL: '1'
    GECKO_HEAD_REV: 5976867c5914929594c7c0d076167ebf254c1695
    MOZ_AUTOMATION: '1'
    TRY_COMMIT_MSG: ''
    MOZ_FETCHES_DIR: fetches
    SCCACHE_DISABLE: '1'
    GECKO_HEAD_REPOSITORY: https://hg.mozilla.org/try
    MOZHARNESS_TEST_PATHS: >-
      {"mochitest-media": ["dom/media/autoplay/test/mochitest/mochitest.ini",
      "dom/media/mediasession/test/mochitest.ini",
      "dom/media/mediasource/test/mochitest.ini",
      "dom/media/test/mochitest.ini",
      "dom/media/test/mochitest_background_video.ini",
      "dom/media/test/mochitest_bugs.ini",
      "dom/media/test/mochitest_compat.ini", "dom/media/test/mochitest_eme.ini",
      "dom/media/test/mochitest_media_recorder.ini",
      "dom/media/test/mochitest_seek.ini",
      "dom/media/test/mochitest_stream.ini",
      "dom/media/webaudio/test/blink/mochitest.ini",
      "dom/media/webaudio/test/mochitest.ini",
      "dom/media/webaudio/test/mochitest_audio.ini",
      "dom/media/webaudio/test/mochitest_bugs.ini",
      "dom/media/webaudio/test/mochitest_media.ini",
      "dom/media/webcodecs/test/mochitest.ini",
      "dom/media/webrtc/tests/mochitests/identity/mochitest.ini",
      "dom/media/webrtc/tests/mochitests/mochitest.ini",
      "dom/media/webrtc/tests/mochitests/mochitest_datachannel.ini",
      "dom/media/webrtc/tests/mochitests/mochitest_getusermedia.ini",
      "dom/media/webrtc/tests/mochitests/mochitest_peerconnection.ini",
      "dom/media/webspeech/recognition/test/mochitest.ini",
      "dom/media/webspeech/synth/test/mochitest.ini",
      "dom/media/webspeech/synth/test/startup/mochitest.ini",
      "dom/media/webvtt/test/mochitest/mochitest.ini"]}
    EXTRA_MOZHARNESS_CONFIG: >-
      {"installer_url":
      "https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/ewVFJ9njQE2zW_vqd50QxQ/artifacts/public/build/target.zip",
      "test_packages_url":
      "https://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/ewVFJ9njQE2zW_vqd50QxQ/artifacts/public/build/target.test_packages.json"}
  mounts:
    - format: zip
      content:
        taskId: ewVFJ9njQE2zW_vqd50QxQ
        artifact: public/build/mozharness.zip
      directory: mozharness
    - file: ./run-task
      content:
        url: >-
          http://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/JrG9cz9SRoKzxkIJPcB0ZA/artifacts/public/run-task
    - file: ./fetch-content
      content:
        url: >-
          http://firefox-ci-tc.services.mozilla.com/api/queue/v1/task/JrG9cz9SRoKzxkIJPcB0ZA/artifacts/public/fetch-content
  command:
    - >-
      C:/mozilla-build/python3/python3.exe run-task --
      c:\mozilla-build\python3\python3.exe -u
      mozharness\scripts\desktop_unittest.py --cfg
      mozharness\configs\unittests\win_unittest.py
      --setpref=webgl.out-of-process=false
      --setpref=security.sandbox.content.headless=false
      --tag=media-engine-compatible
      --setpref=media.wmf.media-engine.enabled=true
      --setpref=media.wmf.media-engine.channel-decoder.enabled=true
      --mochitest-suite=mochitest-media
      --setpref=media.peerconnection.mtransport_process=false
      --setpref=network.process.enabled=false
      --setpref=layers.d3d11.enable-blacklist=false --download-symbols ondemand
  artifacts:
    - name: public/logs
      path: logs
      type: directory
      expires: '2023-05-26T07:11:09.708Z'
    - name: public/test_info
      path: build/blobber_upload_dir
      type: directory
      expires: '2023-05-26T07:11:09.708Z'
  maxRunTime: 5400
  onExitStatus:
    retry:
      - 1073807364
      - 3221225786
      - 4
      - 137
metadata:
  name: test-windows11-64-2009-qr/opt-mochitest-media-wmfme
  owner: bobowencode@gmail.com
  source: >-
    https://hg.mozilla.org/try/file/5976867c5914929594c7c0d076167ebf254c1695/taskcluster/ci/test
  description: >-
    Mochitest media run with Windows Media Foundation media engine enabled
    ([Treeherder
    job](https://treeherder.mozilla.org/#/jobs?repo=try&revision=5976867c5914929594c7c0d076167ebf254c1695&selectedTaskRun=PnVQXgGdSQiwfMkgRCBEng))
tags:
  os: windows
  kind: test
  label: test-windows11-64-2009-qr/opt-mochitest-media-wmfme
  retrigger: 'true'
  test-type: mochitest
  tests_grouped: '1'
  createdForUser: bobowencode@gmail.com
  worker-implementation: generic-worker
extra:
  index:
    rank: 0
  suite: mochitest-media
  chunks:
    total: 1
    current: 1
  parent: JrG9cz9SRoKzxkIJPcB0ZA
  treeherder:
    tier: 2
    symbol: mda
    jobKind: test
    machine:
      platform: windows11-64-2009-qr
    groupName: Mochitests with Windows Media Foundation media engine enabled
    collection:
      opt: true
    groupSymbol: M-wmfme
  test-setting:
    _hash: 6e412562e543
    build:
      type: opt
    runtime:
      wmf-media-engine: true
    platform:
      os:
        name: windows
        build: '2009'
        version: '11'
      arch: '64'
  treeherder-platform: windows11-64-2009-qr/opt
```

**Comment by petemoore** [comment]
And in case try heads get pruned, here are the 4 try push changesets for the try push...

* [patch1.txt](https://github.com/taskcluster/taskcluster/files/11334089/patch1.txt)
* [patch2.txt](https://github.com/taskcluster/taskcluster/files/11334090/patch2.txt)
* [patch3.txt](https://github.com/taskcluster/taskcluster/files/11334091/patch3.txt)
* [patch4.txt](https://github.com/taskcluster/taskcluster/files/11334092/patch4.txt)

File: workers/generic-worker/process/multiuser_windows.go [hunk 1]
```diff
@@ -47,6 +43,7 @@ func TaskUserPlatformData() (pd *PlatformData, err error) {
 		return
 	}
 	pd.CommandAccessToken = pd.LoginInfo.AccessToken()
+	win32.DumpTokenInfo(pd.CommandAccessToken)
```
Comment (lotas) [inline workers/generic-worker/process/multiuser_windows.go:46]: where does it dump token info? to the stdout and into logs directly?

Comment (petemoore) [reply workers/generic-worker/process/multiuser_windows.go:46]: Yes, exactly!

**Review by lotas — APPROVED** [review]
good luck testing this 🤞

**Review by matt-boris — APPROVED** [review]
🤞🏻

**Pull request merged**